### PR TITLE
User activation changes

### PR DESCRIPTION
**Pull request by shahthepro**
### Checklist
- [x] Redirect user to onboard flow when he intends to perform any interaction with DApp
- [ ] Clean up  code

### CODE REVIEW

**Comment by shahthepro** [comment]
@micahalcorn and @nick I have submitted this as a draft PR. However it cannot be merged directly with master due to merge conflicts (since I started this before @nick's meta-tx branch got merged). I'll work on resolving the conflicts and submit the PR for review.

**Comment by micahalcorn** [comment]
I'm going to start a WIP to-do list here. Not all of it will necessarily be blocking.

- [ ] wallet dropdown has just a white bar after connecting MetaMask using account `0x627306090abaB3A6e1400e9345bC60c78a8BEf57`. It should be displaying the default avatar.

**Comment by nick** [comment]
What do you think of breaking up this PR into more manageable chunks as it's a lot of change to introduce in one go. @micahalcorn @shahthepro

**Comment by shahthepro** [comment]
@nick I was hoping to complete and submit the PR this weekend. Let me know how you want it to be split. I can submit multiple small PRs

**Comment by shahthepro** [comment]
@micahalcorn 

Have updated [the test build](https://originprotocol.github.io/test-builds/user-activation/#/). I'm testing the build at present.

> * attempting to add a listing before creating a profile should route to user activation.
>  * attempting to send a message before enabling messaging should route to user activation instead of presenting a modal.

For these two, I check only at the menu items. The user can still load the corresponding URL manually and access those sections. 

> * nav link for wallet dropdown has just a white bar after connecting MetaMask using account 0x627306090abaB3A6e1400e9345bC60c78a8BEf57. It should be displaying the default avatar.
> * nav link for wallet dropdown has just a white bar after creating a profile on Coinbase Wallet. It should be displaying my profile photo.

This was caused since I have not called refetch in the previous build. It is fixed on MetaMask but I couldn't get past "Sign Message" screen on Coinbase since it cannot be run on private network. Will try it on Rinkeby or Ropsten and report back on this.

>  horizontal overflow on small screens when wallet dropdown is open.

Still couldn't figure out why this happens. Will check it out.

>  issues scrolling vertically on step 2

I am using `position: fixed` and there seems to be some known bugs with fixed position elements and Safari (browser/webview) on iOS. I have done a workaround so that it doesn't occur but may have to revisit it.

**Comment by micahalcorn** [comment]
@shahthepro why not validate the listing creation route in that component instead of at each link? I believe that there are various links around the DApp where it would need to happen.

I will be updating this comment with other notes as a review the test build.

- [ ]

**Comment by micahalcorn** [comment]
Round 3️⃣

**Coinbase Wallet**

- [ ] We need to bring back the ability to skip the Origin Rewards signup when editing a profile or adding a verification. Sorry this seems to be missing from Zeplin.
- [ ] I am getting a white page "Error!" when I abandon the name & photo portion of activation. From "Add name and photo", I click the DApp's back arrow and it crashes after about one second. When I reload the page, I am still in the "Add name and photo" state. It may be worth noting that I had previously verified my email address but have NOT published any of it.
- [ ] The recipient in the native confirmation modal now shows "Testing listing upload - Origin" instead of the listing that I wanted to purchase as mentioned previously.
- [ ] If I navigate away from "Add name and photo", I lose my work. This should be retained unless I clear it.

File: dapps/marketplace/src/pages/nav/Profile.js [hunk 1]
```diff
@@ -73,207 +91,332 @@ class ProfileNav extends Component {
...
+  }
+
+  return 'metamask'
+}
+
+export default withWallet(withIdentity(ProfileNav))
 
 require('react-styl')(`
+  #app
+    overflow: hidden !important
```
Comment (nick) [inline dapps/marketplace/src/pages/nav/Profile.js]: seems like a hack... is there no way to find out what's causing the overflow?

Comment (shahthepro) [reply dapps/marketplace/src/pages/nav/Profile.js]: Will look into it again

Comment (shahthepro) [reply dapps/marketplace/src/pages/nav/Profile.js]: The callout arrow was still being rendered despite having 0 height and width. This fixed it: https://github.com/OriginProtocol/origin/pull/2308/commits/456a1fe41ae4896f6fe6b82640a19a1c3c99ee11

**Comment by micahalcorn** [comment]
Great work, @shahthepro 👍 

@nick I think I'm satisfied now. There are a few non-critical issues that I will file, but I don't want to hold this up any longer. Please review and merge when you're ready.

**Comment by nick** [comment]
Getting this warning when I click the mobile dropdown:

image

**Comment by shahthepro** [comment]
@nick 
>  Related... the background image from above is pixelated. Can we use an svg?

Have made the changes you mentioned other than this. I can only find bitmap image on Zeplin. Will ping Aure regarding that.

**Comment by micahalcorn** [comment]
@nick this now handles #2397 as well. ✓

**Pull request merged**